### PR TITLE
feat(linter): add infrastructure for handling early errors

### PR DESCRIPTION
**Pull request by Boshen**
@shannonrothe @jason89521 @YangchenYe323 Please take a look.

This is something *really* cool, we are checking early errors (defined by the spec) in the linter.

This enables us to be like [sucrase](https://github.com/alangpierce/sucrase), where all time consuming checks are moved to the linter, leaving us with a super fast parser. 

This should enable us with some great opportunities ;-)

### CODE REVIEW

File: crates/oxc_linter/src/lib.rs [hunk 1]
```diff
@@ -81,6 +84,7 @@ impl Linter {
         let ctx = LintContext::new(source_text, semantic.clone(), fix);
 
         for node in semantic.nodes().iter() {
+            self.early_error_javascript.run(node, &ctx);
```
Comment (Boshen) [inline crates/oxc_linter/src/lib.rs:87]: This can be toggled, eventually.

File: crates/oxc_linter/src/rules.rs [hunk 2]
```diff
@@ -1,3 +1,7 @@
+pub mod early_error {
+    pub mod javascript;
+}
+
```
Comment (Boshen) [inline crates/oxc_linter/src/rules.rs:4]: This is not part of the lint rules, but it's here for code organization.

File: crates/oxc_linter/src/rules/early_error/javascript.rs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            _ => {}
+        }
+    }
+}
+
+fn check_regexp_literal(lit: &RegExpLiteral, ctx: &LintContext) {
+    #[derive(Debug, Error, Diagnostic)]
+    #[error("The 'u' and 'v' regular expression flags cannot be enabled at the same time")]
+    #[diagnostic()]
+    struct RegExpFlagUAndV(#[label] Span);
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/early_error/javascript.rs:27]: Putting this inside should make things more readable once this file becomes big.

**Review by shannonrothe — APPROVED** [review]
🚀

**Pull request merged**